service/src/main/resources/db/migration/V262__child_absences_function.sql

```diff
@@ -0,0 +1,48 @@
+CREATE FUNCTION absences_in_range(childIds uuid[], period daterange)
+    RETURNS TABLE
+            (
+                id               uuid,
+                child_id         uuid,
+                date             date,
+                absence_type     absence_type,
+                modified_at      timestamp with time zone,
+                modified_by      uuid,
+                category         absence_category,
+                questionnaire_id uuid
+            )
+AS
+$$
+BEGIN
+    RETURN QUERY
+        SELECT *
+        FROM absence a
+        WHERE between_start_and_end($2, a.date)
+          AND ($1 IS NULL OR a.child_id = ANY ($1))
+        UNION
+        SELECT dst.id                             as id,
+               dst.child_id                       as child_id,
+               d::date                            as date,
+               'OTHER_ABSENCE'::absence_type      as absence_type,
+               dst.updated                        as modified_at,
+               null::uuid                         as modified_by,
+               unnest(absence_categories(p.type)) as category,
+               null::uuid                         as questionnaire_id
+        FROM daily_service_time as dst,
+             generate_series(lower($2), upper($2) - 1, '1 day') as d
+                 JOIN placement as p ON d BETWEEN p.start_date AND p.end_date
+        WHERE ($1 IS NULL OR dst.child_id = ANY ($1))
+          AND p.child_id = dst.child_id
```

Comment:
I think the join on placement table is a more appropriate place for this condition.

Reply:
I had that at first, but it complains about it not being possible to reference `dst` in the join's ON clause.

Reply:
Oh yeah, it's because you're not joining the generated series on `daily_service_time`. You might as well do that since you have this `AND dst.validity_period @> d::date` condition in the where clause as well.

Reply:
`ERROR: invalid reference to FROM-clause entry for table "dst"
  Hint: There is an entry for table "dst", but it cannot be referenced from this part of the query.`